libs/messaging/messaging-impl/src/main/kotlin/net/corda/messaging/mediator/TaskManagerHelper.kt

```diff
@@ -0,0 +1,160 @@
+package net.corda.messaging.mediator
+
+import kotlinx.coroutines.runBlocking
+import net.corda.libs.statemanager.api.State
+import net.corda.libs.statemanager.api.StateManager
+import net.corda.messagebus.api.consumer.CordaConsumerRecord
+import net.corda.messaging.api.mediator.MediatorMessage
+import net.corda.messaging.api.mediator.MessageRouter
+import net.corda.messaging.api.mediator.taskmanager.TaskManager
+import net.corda.messaging.api.mediator.taskmanager.TaskType
+import net.corda.messaging.api.processor.StateAndEventProcessor
+import net.corda.messaging.api.records.Record
+import net.corda.messaging.utils.toRecord
+
+/**
+ * Helper that creates and executes various tasks used by [MultiSourceEventMediatorImpl].
+ */
+internal class TaskManagerHelper<K : Any, S : Any, E : Any>(
+    private val taskManager: TaskManager,
+    private val stateManagerHelper: StateManagerHelper<K, S, E>,
+) {
+
+    /** Same as [Callable] but with suspend function call. */
+    fun interface SuspendCallable<V> {
+        @Throws(Exception::class)
+        suspend fun call(): V
+    }
+
+    /**
+     * Creates [ProcessorTask]s for given events and states.
+     *
+     * @param messageGroups Map of messages keys and related events.
+     * @param persistedStates Mpa of message keys and related states.
```

Comment:
```suggestion
     * @param persistedStates Map of message keys and related states.
```
Typo, can be fixed in a future PR